hw/lint/parse-lint-report.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""Parses lint report and dump filtered messages in hjson format.
+"""
+import argparse
+import re
+import mistletoe
+import datetime
+import hjson
+
+from pathlib import Path
+
+def get_results(resdir):
+    """
+    Parse report and corresponding logfiles and extract error, warning
+    and info messages for each IP present in the result folder
+    """
+    results = {"tool": "ascentlint",
+               "errors" : [],
+               "warnings" : [],
+               "lint_errors" : [],
+               "lint_warnings" : [],
+               "lint_infos" : []}
+    try:
+        # check the log file for flow errors and warnings
+        with Path(resdir).joinpath('ascentlint.log').open() as f:
+            full_file = f.read()
+            f.close()
+            results["errors"]  += re.findall(re.compile("^FlexNet Licensing error.*", re.MULTILINE),
+                full_file)
+            results["errors"]  += re.findall(re.compile("^  ERR .*", re.MULTILINE),
+                full_file)
+            results["warnings"] += re.findall(re.compile("^  WARN .*", re.MULTILINE),
+                full_file)
+    except IOError:
+        results["errors"] += ["IOError: could not open tool log"]
+
+    try:
+        # check the report file for lint INFO, WARNING and ERRORs
+        with Path(resdir).joinpath('ascentlint.rpt').open() as f:
+            full_file = f.read()
+            f.close()
+            results["lint_errors"] += re.findall(re.compile("^E  .*", re.MULTILINE),
+                full_file)
+            results["lint_warnings"] += re.findall(re.compile("^W  .*", re.MULTILINE),
+                full_file)
+            results["lint_infos"] += re.findall(re.compile("^I  .*", re.MULTILINE),
+                full_file)
+    except IOError:
+        results["errors"] += ["IOError: could not open tool report"]
+
+    return results
+
+
+def main():
+
+    parser = argparse.ArgumentParser(prog="")
+    parser.add_argument('--resdir',
+                        type=str,
+                        default="./",
+                        help="""""")
+
+    args = parser.parse_args()
+
+    results = get_results(args.resdir)
+
+    result_file = open("results.hjson", "w")
+    hjson.dump(results, result_file, ensure_ascii=False, for_json=True, use_decimal=True)
+    result_file.close()
+
+    # return nonzero status if any warnings or errors are present
+    # lint infos do not count as failures
+    exit(sum([len(results["warnings"]),
+              len(results["errors"]),
+              len(results["lint_warnings"]),
+              len(results["lint_errors"])])>0)
```

Comment:
This should really be more than one statement.

Reply:
you mean this should generate different error codes?

Reply:
One error code is fine. I'd just prefer something like:

```
exit_code = 0
nr_errors = len(results["warnings"]) + .....
print("Lint not successful, got %d errors." % (nr_errors, ))
if nr_errors > 0:
   exit_code = 1
sys.exit(exit_code)
```

Reply:
got it, thanks - added.